Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1

```diff
@@ -0,0 +1,111 @@
+Param(
+    [Parameter(HelpMessage = "The project for which to download dependencies", Mandatory = $true)]
+    [string] $project,
+    [string] $baseFolder,
+    [string] $buildMode = 'Default',
+    [string] $projectsDependenciesJson,
+    [string] $destinationPath,
+    [string] $token
+)
+
+function DownloadDependenciesFromProbingPaths($baseFolder, $project, $destinationPath) {
+    $projectSettings = ReadSettings -baseFolder $baseFolder -project $project
+    
+    $probingPaths = $projectSettings.appDependencyProbingPaths | ConvertFrom-Json
+
+    $downloadedDependencies = @()
+    if ($probingPaths) {
+        $downloadedDependencies += Get-Dependencies -probingPathsJson $repo.appDependencyProbingPaths -saveToPath $destinationPath  | Where-Object { $_ }
+    }
+
+    return $downloadedDependencies
+}
+
+function DownloadDependenciesFromCurrentBuild($baseFolder, $project, $projectsDependencies, $buildMode, $destinationPath) {  
+    Write-Host "Downloading dependencies for project '$project'"
+    
+    $dependencyProjects = @()
+    if ($projectsDependencies.Keys -contains $project) {
+        $dependencyProjects = @($projectsDependencies."$project")
+    }
+
+    Write-Host "Dependency projects: $($dependencyProjects -join ', ')"
+    
+    # For each dependency project, calculate the corresponding probing path
+    $dependeciesProbingPaths = @($dependencyProjects | ForEach-Object {
+        $dependencyProject = $_
+
+        Write-Host "Reading for project '$dependencyProject'"
+        $dependencyProjectSettings = ReadSettings -baseFolder $baseFolder -project $dependencyProject
+    
+        $dependencyBuildMode = $buildMode
+        if(!($dependencyProjectSettings.buildModes -contains $dependencyBuildMode)) {
+            # Download the default build mode if the specified build mode is not supported for the dependency project
+            Write-Host "Build mode '$dependencyBuildMode' is not supported for project '$dependencyProject'. Using the default build mode."
+            $dependencyBuildMode = 'Default';
+        }
+    
+        return @{
+            "release_status" = "thisBuild"
+            "version" = "latest"
+            "buildMode" = $dependencyBuildMode
+            "projects" = $dependencyProject
+            "repo" = "$ENV:GITHUB_SERVER_URL/$ENV:GITHUB_REPOSITORY"
+            "branch" = $ENV:GITHUB_REF_NAME
+            "baseBranch" = $ENV:GITHUB_BASE_REF_NAME
+            "authTokenSecret" = $token
+        }
+    })
+    
+    # For each probing path, download the dependencies
+    $downloadedDependencies = @()
+    $dependeciesProbingPaths | ForEach-Object {
+        $dependency = Get-Dependencies -probingPathsJson $_ -saveToPath $destinationPath | Where-Object { $_ }
+        $downloadedDependencies += $dependency
+    }
+
+    return $downloadedDependencies
+}
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version 2.0
+
+. (Join-Path -Path $PSScriptRoot -ChildPath "..\AL-Go-Helper.ps1" -Resolve)
```

Comment:
Wrap the remaining part of this script in a try catch?

Reply:
That or add the try-catch pattern in the action.yaml as the other actions?

```try {...} catch { Write-Host "::Error::Unexpected error when running action ($($_.Exception.Message.Replace("`r",'').Replace("`n",' ')))"; exit 1 }```